sysutils/salt/Portfile

```diff
@@ -23,56 +23,94 @@ if {$subport eq $name} {
+    variant python36 conflicts python35 python37 description {python-3.6 support} {
+        python.default_version 36
         depends_build       port:py${python.version}-setuptools
 
         depends_lib-append  port:py${python.version}-yaml \
                             port:py${python.version}-jinja2 \
                             port:py${python.version}-msgpack \
-              port:py${python.version}-tornado \
-              port:py${python.version}-zmq
+                            port:py${python.version}-tornado \
+                            port:py${python.version}-zmq \
+                            port:py${python.version}-dateutil \
+                            port:py${python.version}-cryptography \
+                            port:py${python.version}-pycryptodome \
+                            port:py${python.version}-gnupg \
+                            port:py${python.version}-asn1 \
+                            port:py${python.version}-psutil \
+                            port:py${python.version}-mako \
+                            port:py${python.version}-idna \
+                            port:py${python.version}-gitpython \
+                            port:py${python.version}-cherrypy \
+                            port:py${python.version}-cffi \
+                            port:py${python.version}-setproctitle \
+                            port:py${python.version}-pyobjc
 
         destroot.cmd-append --with-salt-version=${version}
     }
 
-    variant python36 conflicts python27 python35 description {python-3.6 support} {
-        python.default_version 36
+    variant python37 conflicts python35 python36 description {python-3.7 support} {
+        python.default_version 37
         depends_build       port:py${python.version}-setuptools
 
         depends_lib-append  port:py${python.version}-yaml \
                             port:py${python.version}-jinja2 \
                             port:py${python.version}-msgpack \
-              port:py${python.version}-tornado \
-              port:py${python.version}-zmq
+                            port:py${python.version}-tornado \
+                            port:py${python.version}-dateutil \
+                            port:py${python.version}-zmq \
+                            port:py${python.version}-pycryptodome \
+                            port:py${python.version}-gnupg \
+                            port:py${python.version}-asn1 \
+                            port:py${python.version}-psutil \
+                            port:py${python.version}-mako \
+                            port:py${python.version}-idna \
+                            port:py${python.version}-gitpython \
+                            port:py${python.version}-cherrypy \
+                            port:py${python.version}-cffi \
+                            port:py${python.version}-setproctitle \
+                            port:py${python.version}-pyobjc
```

Comment:
Theres a lot of duplication in the various python variants. You could probably simplify things by defining them in a foreach loop, so the share the same logic for each.

Reply:
I vaguely recall struggling with the TCL trying to get something DRY to work. I feel that though, and yes that would be preferable.

Reply:
One caveat for that: if there are differences in the dependencies' which require the variants to diverge from each others' build, the DRY libdepends would need to be reverted. I'm not sure which way is more DRY, so I vote we leave it for now and revisit this when time gives us better perspective.